availability/watcher.py

```diff
@@ -141,6 +143,19 @@ def main(period=None):
                   "otherwice threads could crowd round.")
         return 1
 
+    backend = config.get_config().get("backend")
+    if backend["type"] == "elastic":
+        if not backend["connection"]:
+            LOG.error("Elasticsearch is not configured")
+            return 1
+    else:
+        raise NotImplementedError("Unexpected backend: %(type)s" % backend)
+
+    connection = backend["connection"]
+    if not storage.get_elasticsearch([connection]):
+        LOG.error("Failed to set up Elasticsearch")
+        return 1
```

Comment:
Looks like this check is not really useful and can be removed, since es may fail immediately after this line for example. There is no guarantee, that it will be active during the job run.

Also I'd vote for wrapping this function in a decorator and aborting it on any exception from ElasticSearch. smth like https://schedule.readthedocs.io/en/stable/faq.html#what-if-my-task-throws-an-exception